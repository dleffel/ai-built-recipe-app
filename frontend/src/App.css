@import url('https://fonts.googleapis.com/css2?family=Crimson+Pro:ital,wght@0,400;0,600;1,400&display=swap');

:root {
  /* Kindle colors - strictly grayscale */
  --kindle-bg: #F2F2E9;
  --kindle-text: #2C2C2C;
  --kindle-muted: #595959;
  --kindle-border: #D9D9D9;
  --kindle-active: #4D4D4D;
  
  /* Spacing */
  --spacing-1: 0.25rem;
  --spacing-2: 0.5rem;
  --spacing-3: 1rem;
  --spacing-4: 1.5rem;
  --spacing-5: 2rem;
  --spacing-6: 2.5rem;
  
  /* Typography */
  --font-content: 'Crimson Pro', 'Bookerly', Georgia, serif;
  --font-ui: -apple-system, 'Amazon Ember', sans-serif;
  --line-height-content: 1.8;
  --font-size-base: 1.125rem;
  
  /* Layout */
  --page-margin: 1.5rem;
  --content-width: calc(100% - (var(--page-margin) * 2));
}

/* Kindle paper texture */
body {
  margin: 0;
  padding: 0;
  background-color: var(--kindle-bg);
  color: var(--kindle-text);
  font-family: var(--font-content);
  font-size: var(--font-size-base);
  line-height: var(--line-height-content);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  min-height: 100vh;
  position: relative;
}

body::before {
  content: '';
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: var(--kindle-bg);
  opacity: 0.15;
  background-image: url("data:image/svg+xml,%3Csvg viewBox='0 0 200 200' xmlns='http://www.w3.org/2000/svg'%3E%3Cfilter id='noise'%3E%3CfeTurbulence type='fractalNoise' baseFrequency='1.5' numOctaves='5' stitchTiles='stitch'/%3E%3C/filter%3E%3Crect width='100%25' height='100%25' filter='url(%23noise)'/%3E%3C/svg%3E");
  filter: contrast(100%) brightness(100%);
  pointer-events: none;
  z-index: 0;
  mix-blend-mode: multiply;
}

.App {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  position: relative;
  z-index: 1;
  background: transparent;
}

.App-header {
  font-family: var(--font-ui);
  color: var(--kindle-text);
  border-bottom: 1px solid var(--kindle-border);
  position: relative;
  height: 48px;
  padding: 0 var(--page-margin);
  margin-bottom: var(--spacing-4);
  background: transparent;
}

.header-content {
  max-width: var(--max-width);
  margin: 0 auto;
  padding: 0 var(--spacing-3);
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 100%;
  position: relative;
}

.header-left {
  display: flex;
  align-items: center;
  gap: var(--spacing-3);
  height: 100%;
}

.App-header h1 {
  margin: 0;
  font-size: 1.125rem;
  font-weight: 600;
  color: var(--kindle-text);
  line-height: 1;
  font-family: var(--font-content);
}

.breadcrumb {
  color: var(--kindle-muted);
  font-size: 0.875rem;
  display: flex;
  align-items: center;
  gap: var(--spacing-2);
  font-family: var(--font-ui);
}

.breadcrumb span {
  color: var(--kindle-text);
  font-weight: 500;
}

.auth-section {
  display: flex;
  align-items: center;
  gap: var(--spacing-3);
  padding: var(--spacing-1) 0;
  height: 100%;
}

.App-main {
  flex: 1;
  width: 100%;
  animation: fadeIn 0.3s ease-out;
  box-sizing: border-box;
  background: transparent;
}

.loading {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  color: var(--kindle-muted);
  font-size: var(--font-size-base);
  font-family: var(--font-ui);
  font-style: italic;
}

.login-message {
  text-align: center;
  padding: var(--spacing-6) var(--page-margin);
  color: var(--kindle-text);
  font-size: var(--font-size-base);
  max-width: var(--content-width);
  margin: var(--spacing-6) auto;
  font-family: var(--font-content);
  line-height: var(--line-height-content);
}

.welcome-message {
  text-align: center;
  margin-bottom: var(--spacing-5);
  color: var(--kindle-text);
  font-size: 1.75rem;
  font-weight: 600;
  line-height: 1.3;
  font-family: var(--font-content);
  animation: slideDown 0.5s ease-out;
}

.recipe-container {
  width: 100%;
  padding: 0;
  box-sizing: border-box;
  background: transparent;
}

.recipe-header {
  display: flex;
  justify-content: flex-end;
  padding: 0 var(--page-margin);
  margin-bottom: var(--spacing-4);
  width: 100%;
  box-sizing: border-box;
  background: transparent;
}

.create-button {
  padding: var(--spacing-3) var(--spacing-4);
  background: transparent;
  color: var(--kindle-text);
  border: 1px solid var(--kindle-border);
  font-size: var(--font-size-base);
  font-weight: 500;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  display: inline-flex;
  align-items: center;
  gap: var(--spacing-2);
  font-family: var(--font-ui);
}

.create-button:hover {
  color: var(--kindle-active);
  border-color: var(--kindle-active);
}

.create-button:active {
  background-color: var(--kindle-border);
  color: var(--kindle-active);
}

@keyframes fadeIn {
  from { opacity: 0; }
  to { opacity: 1; }
}

@keyframes slideDown {
  from {
    opacity: 0;
    transform: translateY(-20px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

@keyframes pulse {
  0% { opacity: 1; }
  50% { opacity: 0.5; }
  100% { opacity: 1; }
}

@media (max-width: 768px) {
  .header-content {
    padding: 0;
  }

  .App-header {
    padding: 0 var(--page-margin);
  }

  .App-header h1 {
    font-size: 1.125rem;
    color: var(--kindle-text);
  }

  .breadcrumb {
    display: none;
  }

  .App-main {
    width: 100%;
    padding: 0;
  }

  .recipe-container {
    padding: 0;
    margin-bottom: var(--spacing-4);
    width: 100%;
  }

  .recipe-header {
    justify-content: center;
    padding: var(--spacing-3) var(--page-margin);
    margin: 0;
  }

  .create-button {
    width: 100%;
    justify-content: center;
    padding: var(--spacing-2) var(--spacing-3);
    font-size: var(--font-size-base);
    font-family: var(--font-ui);
  }

  .login-message {
    margin: var(--spacing-4) var(--page-margin);
    padding: 0;
    width: auto;
    font-size: var(--font-size-base);
  }

  .welcome-message {
    font-size: 1.5rem;
    margin: var(--spacing-4) var(--page-margin);
    width: auto;
  }
}
