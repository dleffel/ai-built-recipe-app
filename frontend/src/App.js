import React from 'react';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>Recipe App</h1>
        <p>Welcome to your recipe collection!</p>
      </header>
      <main>
        {/* Recipe components will be added here */}
      </main>
    </div>
  );
}

export default App;
