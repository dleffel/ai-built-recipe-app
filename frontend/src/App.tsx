import React, { useState } from 'react';
import './App.css';
import { AuthProvider, useAuth } from './context/AuthContext';
import Login from './components/Login';
import { RecipeList } from './components/RecipeList';
import { RecipeDetail } from './components/RecipeDetail';
import { RecipeForm } from './components/RecipeForm';
import { Recipe, CreateRecipeDTO } from './types/recipe';
import { recipeApi } from './services/api';

type View = 'list' | 'detail' | 'create' | 'edit';

interface ViewState {
  type: View;
  recipeId?: string;
}

const AppContent: React.FC = () => {
  const { user, loading } = useAuth();
  const [currentView, setCurrentView] = useState<ViewState>({ type: 'list' });
  const [selectedRecipe, setSelectedRecipe] = useState<Recipe | null>(null);

  const handleCreateRecipe = async (data: CreateRecipeDTO) => {
    try {
      await recipeApi.create(data);
      setCurrentView({ type: 'list' });
    } catch (error) {
      console.error('Error creating recipe:', error);
    }
  };

  const handleEditRecipe = async (data: CreateRecipeDTO) => {
    if (!selectedRecipe) return;
    
    try {
      await recipeApi.update(selectedRecipe.id, data);
      setCurrentView({ type: 'list' });
    } catch (error) {
      console.error('Error updating recipe:', error);
    }
  };

  const handleDeleteRecipe = async (recipe: Recipe) => {
    try {
      await recipeApi.delete(recipe.id);
      setCurrentView({ type: 'list' });
    } catch (error) {
      console.error('Error deleting recipe:', error);
    }
  };

  const renderContent = () => {
    if (!user) {
      return (
        <div className="login-message">
          <p>Please sign in to access your recipes</p>
        </div>
      );
    }

    switch (currentView.type) {
      case 'detail':
        if (!selectedRecipe) return null;
        return (
          <RecipeDetail
            recipe={selectedRecipe}
            onEdit={() => setCurrentView({ type: 'edit' })}
            onDelete={() => handleDeleteRecipe(selectedRecipe)}
            onBack={() => setCurrentView({ type: 'list' })}
          />
        );

      case 'create':
        return (
          <RecipeForm
            onSubmit={handleCreateRecipe}
            onCancel={() => setCurrentView({ type: 'list' })}
          />
        );

      case 'edit':
        if (!selectedRecipe) return null;
        return (
          <RecipeForm
            recipe={selectedRecipe}
            onSubmit={handleEditRecipe}
            onCancel={() => setCurrentView({ type: 'list' })}
          />
        );

      case 'list':
      default:
        return (
          <div className="recipe-container">
            <div className="recipe-header">
              <h2>My Recipes</h2>
              <button
                onClick={() => setCurrentView({ type: 'create' })}
                className="create-button"
              >
                Create New Recipe
              </button>
            </div>
            <RecipeList
              onRecipeClick={(recipe) => {
                setSelectedRecipe(recipe);
                setCurrentView({ type: 'detail' });
              }}
              onRecipeEdit={(recipe) => {
                setSelectedRecipe(recipe);
                setCurrentView({ type: 'edit' });
              }}
              onRecipeDelete={handleDeleteRecipe}
            />
          </div>
        );
    }
  };

  if (loading) {
    return <div className="loading">Loading...</div>;
  }

  return (
    <div className="App">
      <header className="App-header" role="banner">
        <div className="header-content">
          <div className="header-left">
            <h1>Recipe App</h1>
            {currentView.type !== 'list' && (
              <div className="breadcrumb" role="navigation">
                <span
                  onClick={() => setCurrentView({ type: 'list' })}
                  style={{ cursor: 'pointer' }}
                >
                  My Recipes
                </span>
                <span>/</span>
                <span>
                  {currentView.type === 'detail' ? selectedRecipe?.title :
                   currentView.type === 'create' ? 'New Recipe' :
                   currentView.type === 'edit' ? `Edit ${selectedRecipe?.title}` : ''}
                </span>
              </div>
            )}
          </div>
          <Login />
        </div>
      </header>
      <main className="App-main">
        {renderContent()}
      </main>
    </div>
  );
};

const App: React.FC = () => {
  return (
    <AuthProvider>
      <AppContent />
    </AuthProvider>
  );
};

export default App;