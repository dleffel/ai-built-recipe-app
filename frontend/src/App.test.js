import { render, screen } from '@testing-library/react';
import App from './App';

describe('App Component', () => {
  it('renders without crashing', () => {
    render(<App />);
    expect(screen.getByRole('heading', { name: 'Recipe App' })).toBeInTheDocument();
    expect(screen.getByText('Welcome to your recipe collection!')).toBeInTheDocument();
  });

  it('has correct header styling', () => {
    render(<App />);
    const header = screen.getByRole('banner');
    
    expect(header).toHaveClass('App-header');
    expect(header).toBeVisible();
  });

  it('matches snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
