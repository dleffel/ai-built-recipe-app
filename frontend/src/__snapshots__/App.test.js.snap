// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
      role="banner"
    >
      <h1>
        Recipe App
      </h1>
      <p>
        Welcome to your recipe collection!
      </p>
    </header>
    <main />
  </div>
</div>
`;